build.rs

```diff
@@ -142,3 +92,98 @@ fn main() -> Result<()> {
 
     Ok(())
 }
+
+/// Build the `secp256k1` library.
+fn build_secp256k1() {
+    let mut build = cc::Build::new();
+
+    // Compile C99 code
+    language_std(&mut build, "c99");
+
+    // Define configuration constants
+    build
+        .define("SECP256K1_BUILD", "1")
+        .define("USE_NUM_NONE", "1")
+        .define("USE_FIELD_INV_BUILTIN", "1")
+        .define("USE_SCALAR_INV_BUILTIN", "1")
+        .define("ECMULT_WINDOW_SIZE", "15")
+        .define("ECMULT_GEN_PREC_BITS", "4")
+        // Use the endomorphism optimization now that the patents have expired.
+        .define("USE_ENDOMORPHISM", "1")
+        // Technically libconsensus doesn't require the recovery feature, but `pubkey.cpp` does.
+        .define("ENABLE_MODULE_RECOVERY", "1")
+        // The source files look for headers inside an `include` sub-directory
+        .include("depend/zcash/src/secp256k1")
+        // Some ecmult stuff is defined but not used upstream
+        .flag_if_supported("-Wno-unused-function")
+        .flag_if_supported("-Wno-unused-parameter");
```

Comment:
Do we need `/wd4100` here as well?

Reply:
I don't think so. At least there wasn't anything from CI :man_shrugging: